rpc/src/main/scala/RPCAsyncImplicits.scala

```diff
@@ -58,19 +83,32 @@ trait RPCAsyncImplicits extends freestyle.async.Implicits {
         fa.map(f)
     }
 
-  implicit def task2Future(implicit S: Scheduler): FSHandler[Task, Future] =
+  implicit def task2Future(implicit S: Scheduler): Task ~> Future =
     new TaskMHandler[Future]
 
-  implicit val future2Task: Future ~> Task =
-    new (Future ~> Task) {
-      override def apply[A](fa: Future[A]): Task[A] = {
-        asyncLogger.info(s"${Thread.currentThread().getName} Deferring Future to Task...")
-        Task.deferFuture(fa)
-      }
-    }
-
   implicit val task2Task: Task ~> Task = new (Task ~> Task) {
     override def apply[A](fa: Task[A]): Task[A] = fa
   }
 
+  implicit def task2IO(implicit S: Scheduler): Task ~> IO = new (Task ~> IO) {
+    override def apply[A](fa: Task[A]): IO[A] = fa.toIO
+  }
 }
+
+trait IOAsyncInstances extends BaseAsync {
+
+  implicit val ioComonad: Comonad[IO] = new Comonad[IO] {
```

Comment:
We should get rid of this once we have Streams as first class construct and not nested under `FS[Ob]`

Reply:
Looking forward to see it it too, just a little bit of time to tackle it.